File: test/data/typescript/test06/Statelong2.ts
import {BaseState} from "./BaseState"
import {IControllee} from "./IControllee"
import {StateController} from "./StateController"
export class Statelong2 extends BaseState
{
    private _stateController: StateController;
    private _controlleeImp: IControllee;
    public constructor(stateController: StateController, controllee: IControllee)
    {
        super(controllee);
        this._stateController = stateController;
        this._controlleeImp = controllee;
    }
    public TransitEnoughData(): BaseState | null
    {
        this._stateController.InstanceOfProcessData.Setup(false, false);
        return this._stateController.InstanceOfProcessData;
    }
    public TransitNewData(): BaseState | null
    {
        this._stateController.InstanceOflong2.Setup(false, false);
        return this._stateController.InstanceOflong2;
    }
    public  GetStateName(): string
    {
        return "State3.State4.long2";
    }
    public GetParent(): BaseState | null
    {
        return this._stateController.InstanceOfState4;
    }
}

File: test/data/cpp/test06/Statelong2.cpp
#include "Statelong2.h"
Statelong2::Statelong2(StateController* stateController, IControllee* controllee) : BaseState(controllee)
{
    _stateController = stateController;
    _controllee = controllee;
}
Statelong2::~Statelong2()
{
}
BaseState* Statelong2::TransitEnoughData()
{
    _stateController->InstanceOfProcessData->Setup(false, false);
    return _stateController->InstanceOfProcessData;
}
BaseState* Statelong2::TransitNewData()
{
    _stateController->InstanceOflong2->Setup(false, false);
    return _stateController->InstanceOflong2;
}
const char* Statelong2::GetStateName()
{
    return "State3.State4.long2";
}
BaseState* Statelong2::GetParent()
{
    return _stateController->InstanceOfState4;
}
int Statelong2::GetStateID()
{
    return 6;
}